sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3982,6 +3982,14 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
       messageParameters = Map("provider" -> name))
   }
 
+  def externalDataSourceException(cause: Throwable): Throwable = {
+    new AnalysisException(
+      errorClass = "DATA_SOURCE_EXTERNAL_ERROR",
+      messageParameters = Map(),
```

Comment:
Can we provide any info about the datasource (like name, path and so on) which causes the issue. Just wonder how users should troubleshoot the issue.

Reply:
the idea is the data source fails, so we don't really have any of that (it's either null or non-existant)